crates/wasm-smith/src/lib.rs

```diff
@@ -1076,29 +1193,16 @@ where
     }
 
     fn arbitrary_start(&mut self, u: &mut Unstructured) -> Result<()> {
-        let mut choices = Vec::with_capacity(self.total_funcs as usize);
-        let mut func_index = 0;
+        let mut choices = Vec::with_capacity(self.funcs.len() as usize);
 
         if !self.config.allow_start_export() {
             return Ok(());
         }
```

Comment:
Only just noticed this, but can we put `choices` after the check for whether a start is allowed? Especially since we're eagerly reserving capacity.